tests/system_tests/conv1d/conv1d.py

```diff
@@ -0,0 +1,186 @@
+    vhdl_dir = "./tests/system_tests/conv1d/build_dir"
+    binfile_dir = "./tests/system_tests/conv1d/build_dir_output"
+    binfile_path = Path(binfile_dir + "/output/env5_top_reconfig.bin")
+    skeleton_id = list(range(15, -1, -1))
+    skeleton_id_as_bytearray = bytearray()
+    for x in skeleton_id:
+        skeleton_id_as_bytearray.extend(
+            x.to_bytes(length=1, byteorder="little", signed=False)
+        )
+
+    total_bits = 8
+    frac_bits = 2
+    num_inputs = 5
+    kernel_size = 3
+    num_in_channels = 1
+    num_out_channels = 1
+    # batches = 2
+    num_outputs = num_inputs - kernel_size + 1
+    nn = create_vhd_files(
+        output_dir=vhdl_dir,
+        num_inputs=num_inputs,
+        num_outputs=num_outputs,
+        num_in_channels=num_in_channels,
+        num_out_channels=num_out_channels,
+        kernel_size=kernel_size,
+        total_bits=total_bits,
+        frac_bits=frac_bits,
+        skeleton_id=skeleton_id,
+    )
+
+    fxp_conf = FixedPointConfig(total_bits, frac_bits)
+
+    # inputs = fxp_conf.as_rational(fxp_conf.as_integer(torch.rand(batches, num_in_channels, num_inputs)))
+    inputs = fxp_conf.as_rational(
+        fxp_conf.as_integer(
+            torch.Tensor(
+                [
+                    [[0.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 1.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 1.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 1.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 0.0, 1.0]],
+                    [[2.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 1.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 1.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 1.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 0.0, 1.0]],
+                    [[3.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 2.0, 0.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 2.0, 0.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 2.0, 0.0]],
+                    [[1.0, 0.0, 0.0, 0.0, 2.0]],
+                    [[1.0, 1.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 1.0, 1.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 1.0, 1.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 1.0, 1.0]],
+                    [[-1.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, -1.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, -1.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 0.0, -1.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 0.0, -1.0]],
+                    [[2.0, 0.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 2.0, 0.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 2.0, 0.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 2.0, 0.0]],
+                    [[0.0, 0.0, 0.0, 0.0, 2.0]],
+                ]
+            )
+        )
+    )
+    batches = inputs.shape[0]
+
+    expected_outputs = nn(inputs)
+
+    # vivado_build_binfile(vhdl_dir, binfile_dir)
+
+    serial_con = serial.Serial("/dev/tty.usbmodem1101")
```

Comment:
As this will fail in many cases, we could consider reading the value from an environment variable or from command line arguments. That way we at least would not have to change that line every time the path changes.

No idea how to pass command line flags through pytest, so maybe the environment variable is the way to go here.

Reply:
I am not sure if there will ever be a good way... 
But I guess trying out some stuff is a good starting point. For now I would leave it like it is to make the features available.